sound/soc/sof/sof-client-ipc-test.c

```diff
@@ -31,6 +31,32 @@ struct sof_ipc_client_data {
 	char *buf;
 };
 
+static int sof_ipc_dfsentry_open(struct inode *inode, struct file *file)
+{
+	struct sof_client_dev *cdev;
+	int ret;
+
+	ret = debugfs_file_get(file->f_path.dentry);
```

Comment:
can the debugfs file be removed while open if we already prevent the core from being removed? Or is debugfs managed by a different module?

Reply:
I have tried it with the dma-trace and removed the `sof_client_core_module_get/put` and left only the `debugfs_file_get/put`:
- the `snd_sof_dma_trace` can not be removed as it's use count is increases
- the `snd_sof_pci_intel_tgl` can be removed (use count is 0) and by doing that the `sof-logger` aborts and the dma-trace is unloading fine.

I need to recompile the firmware with debug enabled to be able to test how the ipc-flood would behave, but I kind of expect similar behavior, likely the tgl rmmod would wait until the test finishes.

Reply:
so is this patch needed or not?

Reply:
It is needed, but the `sof_client_core_module_get/put` canm be dropped from it.